src/main/java/org/spongepowered/api/item/inventory/ItemStackComparators.java

```diff
@@ -82,46 +65,44 @@
     public static final Comparator<ItemStack> TYPE_SIZE = Ordering.compound(Arrays.asList(TYPE, SIZE));
 
     /**
-     * Compares ItemStacks based on {@link org.spongepowered.api.item.ItemType},
-     * {@link ItemStack} damage and {@link ItemStack} size. This comparator will
-     * not return the same results as ItemStack.equals(ItemStack) for ItemStacks
-     * with extra attached data.
+     * The default comparator for {@link ItemStack}s.
      */
-    @SuppressWarnings("unchecked")
-    public static final Comparator<ItemStack> TYPE_DAMAGE_SIZE = Ordering.compound(Arrays.asList(TYPE, DAMAGE, SIZE));
-
-    /**
-     * The default comparator for {@link org.spongepowered.api.item.inventory.ItemStack}s.
-     */
-    public static final Comparator<ItemStack> DEFAULT = TYPE_DAMAGE_SIZE;
+    public static final Comparator<ItemStack> DEFAULT = TYPE_SIZE;
 
     private ItemStackComparators() {}
 
     static final class Type implements Comparator<ItemStack> {
 
         @Override
-        public int compare(final ItemStack o1, final ItemStack o2) {
+        public int compare(@Nullable final ItemStack o1, @Nullable final ItemStack o2) {
+            if (o1 == null && o2 == null) {
+                return 0;
+            }
+            if (o1 == null) {
+                return 1;
+            }
+            if (o2 == null) {
+                return -1;
+            }
             return o1.getItem().getId().compareTo(o2.getItem().getId());
         }
 
     }
 
-    static final class Damage implements Comparator<ItemStack> {
-
-        @Override
-        public int compare(final ItemStack o1, final ItemStack o2) {
-            return o1.getDamage() - o2.getDamage();
-        }
-
-    }
-
     static final class Size implements Comparator<ItemStack> {
 
         @Override
-        public int compare(final ItemStack o1, final ItemStack o2) {
+        public int compare(@Nullable final ItemStack o1, @Nullable final ItemStack o2) {
+            if (o1 == null && o2 == null) {
+                return 0;
+            }
+            if (o1 == null) {
+                return 1;
+            }
+            if (o2 == null) {
+                return -1;
+            }
             return o1.getQuantity() - o2.getQuantity();
         }
-
     }
-
```

Comment:
Note that comparing individual `ItemData` objects is already supported on their own. I'll add comparators to check whether both item stacks have all of the same (properties, item data, type etc.)